src/CommunityToolkit.Mvvm.CodeFixers/ClassUsingAttributeInsteadOfInheritanceCodeFixer.cs

```diff
@@ -76,21 +77,17 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
     /// Applies the code fix to a target class declaration and returns an updated document.
     /// </summary>
     /// <param name="document">The original document being fixed.</param>
+    /// <param name="root">The original tree root belonging to the current document.</param>
     /// <param name="classDeclaration">The <see cref="ClassDeclarationSyntax"/> to update.</param>
     /// <param name="attributeTypeName">The name of the attribute that should be removed.</param>
     /// <param name="cancellationToken">The cancellation token for the operation.</param>
     /// <returns>An updated document with the applied code fix, and <paramref name="classDeclaration"/> inheriting from <c>ObservableObject</c>.</returns>
-    private static async Task<Document> UpdateReference(Document document, ClassDeclarationSyntax classDeclaration, string attributeTypeName, CancellationToken cancellationToken)
+    private static Task<Document> UpdateReference(Document document, SyntaxNode root, ClassDeclarationSyntax classDeclaration, string attributeTypeName, CancellationToken cancellationToken)
     {
         // Insert ObservableObject always in first position in the base list. The type might have
         // some interfaces in the base list, so we just copy them back after ObservableObject.
-        ClassDeclarationSyntax updatedClassDeclaration =
-            classDeclaration.WithBaseList(BaseList(SingletonSeparatedList(
-            (BaseTypeSyntax)SimpleBaseType(IdentifierName("ObservableObject"))))
-            .AddTypes(classDeclaration.BaseList?.Types.ToArray() ?? Array.Empty<BaseTypeSyntax>()));
-
-        AttributeListSyntax? targetAttributeList = null;
-        AttributeSyntax? targetAttribute = null;
+        SyntaxGenerator generator = SyntaxGenerator.GetGenerator(document);
+        ClassDeclarationSyntax updatedClassDeclaration = (ClassDeclarationSyntax)generator.AddBaseType(classDeclaration, IdentifierName("ObservableObject"));
```

Comment:
Note: what if there is no `using` for the containing namespace of `ObservableObject`?

Reply:
Don't think we have to, as it's in the same namespace as the attribute being removed. I assume in theory one could have used the attribute by fully qualified type name instead, but not sure it's worth worrying about 🤔